autoemulate/core/results.py

```diff
@@ -196,3 +197,23 @@ def get_result(self, result_id: int) -> Result:
             return self._id_to_result[result_id]
         except KeyError as err:
             raise ValueError(f"No result found with ID: {result_id!s}") from err
+
+    def get_result_by_model_name(self, model_name: str) -> Result:
```

Comment:
Presumably model name here is, e.g. 'gp'. What happens if there are multiple GPs trained (e.g., with different transforms)?

Reply:
As discussed - it would be good to address in a new issue in more detail but for now it should:
- raise exception if more than one model with the same name for multiple transforms
- use `get_model_by_id` in the `fit_from_reinitialized` method